rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -1750,10 +1702,9 @@ extern "C"
             {
                 void *data = guard_conditions->guard_conditions[i];
                 GuardCondition *guard_condition = (GuardCondition*)data;
-                if(!guard_condition->getHasTriggered())
-                {
-                    guard_conditions->guard_conditions[i] = 0;
-                }
+                // Guard conditions persist (don't set to null)
+                // We call getHasTriggered to reset the trigger state.
```

Comment:
At this point, I don't know if we've reached a consensus or not. My vote is to keep setting non-triggered guard conditions to null (and refilling them each loop) and just ignore which ones were triggered until we need that information for whatever reason. But I'm also fine if we take it out now.

@jacquelinekay I don't know if you just want to make a decision and run with it or wait until the next ROS 2 meeting.